spyder/plugins/application/plugin.py

```diff
@@ -326,7 +327,7 @@ def apply_settings(self):
         self._main.apply_settings()
 
     @Slot()
-    def restart(self):
+    def restart(self, reset=False, close_immediately=False):
```

Comment:
`reset` is not used in this method. Should it be removed? It's also mentioned in the docstring, so it should be removed from it too.